experiment/simulation/js/layout.js

```diff
@@ -0,0 +1,130 @@
+import { simulate, deleteElement } from "./gate.js";
+import {
+  connectGate,
+  unbindEvent,
+  initMultiplier,
+  refreshWorkingArea,
+} from "./main.js";
+
+"use strict";
+// Wires Colours
+export const wireColours = [
+  "#ff0000",
+  "#00ff00",
+  "#0000ff",
+  "#bf6be3",
+  "#ff00ff",
+  "#00ffff",
+  "#ff8000",
+  "#00ff80",
+  "#80ff00",
+  "#ff0080",
+  "#8080ff",
+  "#c0c0c0",
+];
+
+// Contextmenu
+const menu = document.querySelector(".menu");
+const menuOption = document.querySelector(".menu-option");
+let menuVisible = false;
+
+const toggleMenu = (command) => {
+  menu.style.display = command === "show" ? "block" : "none";
+  menuVisible = !menuVisible;
+};
+
+export const setPosition = ({ top, left }) => {
+  menu.style.left = `${left}px`;
+  menu.style.top = `${top}px`;
+  toggleMenu("show");
+};
+
+window.addEventListener("click", (e) => {
+  if (menuVisible) toggleMenu("hide");
+  window.selectedComponent = null;
+  window.componentType = null;
+});
+
+menuOption.addEventListener("click", (e) => {
+  if (e.target.innerHTML === "Delete") {
+    if (window.componentType === "gate") {
+      deleteElement(window.selectedComponent);
+    }
+  }
+  window.selectedComponent = null;
+  window.componentType = null;
+});
+
+// Tabs
+
+function changeTabs(e) {
+  const task = e.target.parentNode.id;
+  if (window.currentTab === task) {
+    return;
+  }
+
+  if (window.currentTab !== null) {
+    document.getElementById(window.currentTab).classList.remove("is-active");
+  }
+  window.currentTab = task;
+  document.getElementById(task).classList.add("is-active");
+
+  // Half adder
+  unbindEvent();
+  connectGate();
+  refreshWorkingArea();
+  initMultiplier();
+  window.simulate = simulate;
+  updateToolbar();
+  clearObservations();
+  resize();
+}
+
+window.changeTabs = changeTabs;
+
+// Toolbar
+
+function updateToolbar() {
+  let elem = "";
+  elem =

```

Comment:
Why do we need 2 statements here to assign the value to elem?

Reply:
removed the redundant statement